src/stores/WifiSetupStore.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import type { Navigation, WifiNetwork } from '../types';
+
+import { action, computed, observable, reaction } from 'mobx';
+import {
+  WifiConfigureStore,
+  WifiConnectStore,
+} from './ApiRequestStores/SoftApApiStores';
+import SoftApService from '../SoftApService';
+import { LoadObject } from 'brewskey.js-api';
+import { waitForLoaded } from './DAOStores';
+
+type WifiSetupStep = 1 | 2 | 3 | 4;
+
+class WifiSetupStore {
+  @observable wifiSetupStep: WifiSetupStep = 1;
+  @observable particleID: string = '';
+  @observable _wifiSetupLoaderID = null;
+
+  constructor(navigation: Navigation) {
+    reaction(
+      () => this.wifiSetupStep,
+      (wifiSetupStep: WifiSetupStep, currentReaction: Object) => {
+        // todo make replace instead navigate to prevent using hardware back button
+        navigation.navigate(`wifiSetupStep${wifiSetupStep}`);
+        if (wifiSetupStep === 4) {
+          currentReaction.dispose();
+        }
+      },
+    );
+  }
+
+  @computed
+  get wifiSetupLoader(): LoadObject<void> {
+    return this._wifiSetupLoaderID
+      ? WifiConfigureStore.getFromCache(this._wifiSetupLoaderID).map(
+          (): LoadObject<void> => WifiConnectStore.get(),
+        )
+      : LoadObject.empty();
+  }
+
+  @action
+  onStep1Ready = () => {
+    this.setWifiSetupStep(2);
+    this._queryParticleID();
+  };
+
+  @action
+  setParticleID = (particleID: string) => {
+    this.particleID = particleID;
+  };
+
+  @action
+  setWifiSetupStep = (step: WifiSetupStep) => {
+    this.wifiSetupStep = step;
+  };
+
+  @action
+  setupWifi = async (wifiNetwork: WifiNetwork = {}): Promise<void> => {
+    try {
+      this._wifiSetupLoaderID = WifiConfigureStore.fetch(wifiNetwork);
+      await waitForLoaded(() => this.wifiSetupLoader, 15000);
+      this.setWifiSetupStep(4);
+      // eslint-disable-next-line no-empty
+    } catch (error) {}
```

Comment:
Hmm... if there is an error we should show it in the UI.

Reply:
instead of this try,catch and waitForLoaded I can use one time reaction, but I thought the flow will be less straightforward then..what do you think?

Reply:
I just don't want to swallow errors. I don't understand the full context of this code so maybe it's ok?

Reply:
yeah..that error is not swallowed .. try/catch more like if/else here. If waitForLoded resolved- we do navigation to next route, if not..we don't and use wifiSetupLoader to show up the error in WifiListItem.
Generally..I don't like that flow, as you, either..because its unclear why that all for...but I didn't come up with anything better yet...

Reply:
I think for now I'll end up with reaction instead of this..someting like `when(wifiSetupLoader, () => if(wifiSetupLoader.hasValue()) { navigate})` - this will be more clear